glslc/src/file_compiler.cc

```diff
@@ -158,6 +188,7 @@ void FileCompiler::SetIndividualCompilationFlag() {
 
 void FileCompiler::SetDisassemblyFlag() {
   if (!preprocess_only_) {
+    output_type_ = OutputType::SpirvAssemblyText;
     disassemble_ = true;
```

Comment:
Removed in next commit.
For preprocess_only_ I made a small helper method PreprocessOnly that checks the value of output_type_ appropriately.